File: Bindings~/java/src/test/java/ro/altom/altunitytester/MockNotificationCallBacks.java
package ro.altom.altunitytester;

import ro.altom.altunitytester.Logging.AltUnityLogLevel;
import ro.altom.altunitytester.Notifications.AltUnityLoadSceneNotificationResultParams;
import ro.altom.altunitytester.Notifications.AltUnityLogNotificationResultParams;
import ro.altom.altunitytester.Notifications.BaseNotificationCallbacks;

public class MockNotificationCallBacks extends BaseNotificationCallbacks {

    public static String lastLoadedScene;
    public static String lastUnloadedScene;
    public static String logMessage;
    public static String logStackTrace;
    public static AltUnityLogLevel logLevel = AltUnityLogLevel.Error;
    public static boolean applicationPaused;

    @Override
    public void SceneLoadedCallBack(
            AltUnityLoadSceneNotificationResultParams altUnityLoadSceneNotificationResultParams) {
        lastLoadedScene = altUnityLoadSceneNotificationResultParams.sceneName;
    }
    
    @Override
    public void SceneUnloadedCallBack(String sceneName) {
        lastUnloadedScene = sceneName;
    }

    @Override
    public void LogCallBack(AltUnityLogNotificationResultParams altUnityLogNotificationResultParams) {
        logMessage = altUnityLogNotificationResultParams.message;
        logStackTrace = altUnityLogNotificationResultParams.stackTrace;
        logLevel = AltUnityLogLevel.values()[altUnityLogNotificationResultParams.level];
    }

    @Override
    public void ApplicationPausedCallBack(boolean applicationPaused) {
        MockNotificationCallBacks.applicationPaused = applicationPaused;
    }
}

File: Assets/AltUnityTester/Examples/Test/Editor/Driver/MockClasses/MockNotificationCallBacks.cs
using Altom.AltUnityDriver.Notifications;
namespace Altom.AltUnityDriver.MockClasses
{
    internal class MockNotificationCallBacks : INotificationCallbacks
    {
        public static string LastSceneLoaded = "";
        public static string LastSceneUnloaded = "";
        public static bool ApplicationPaused = false;
        public void SceneLoadedCallback(AltUnityLoadSceneNotificationResultParams altUnityLoadSceneNotificationResultParams)
        {
            LastSceneLoaded = altUnityLoadSceneNotificationResultParams.sceneName;
        }
        public void SceneUnloadedCallback(string sceneName)
        {
            LastSceneUnloaded = sceneName;
        }
        public void ApplicationPausedCallback(bool applicationPaused)
        {
            ApplicationPaused = applicationPaused;
        }
    }
}
